grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -475,15 +475,27 @@ class ProductAvailabilityService {
     }
 
     def getInventoryByProduct(Location location) {
+        return getInventoryByProduct(location, [])
+    }
+
+    def getInventoryByProduct(Location location, List<Category> categories) {
+        def categoriesQuery = "";
+        def queryArguments = [location: location]
+
+        if (categories) {
+            categoriesQuery = "and pa.product.category.id in (:categories)"
+            queryArguments += [categories: categories.id]
```

Comment:
In the future, we should make this explicit just in case something changes. `categories.id` implies `categories*.id` which implies `categories.collect { it.id }`. I would prefer to be explicit.